code/game/gamemodes/nuclear/nuclear_challenge.dm

```diff
@@ -17,17 +17,32 @@
 		return
 
 	declaring_war = 1
-	var/are_you_sure = alert(user, "Consult your team carefully before you declare war on [station_name()]]. Are you sure you want to alert the enemy crew?", "Declare war?", "Yes", "No")
+	var/are_you_sure = alert(user, "Consult your team carefully before you declare war on [station_name()]]. Are you sure you want to alert the enemy crew? You have [round((world.time-round_start_time - CHALLENGE_TIME_LIMIT)/10)] seconds to decide", "Declare war?", "Yes", "No")
 	declaring_war = 0
+
+	if(!check_allowed(user))
+		return
+
 	if(are_you_sure == "No")
 		user << "On second thought, the element of surprise isn't so bad after all."
 		return
+
+	var/war_declaration = "[user.real_name] has declared his intent to utterly destroy [station_name()] with a nuclear device, and dares the crew to try and stop them."
+	var/custom_threat = alert(user, "Do you want to customize your declaration?", "Customize?", "Yes", "No")
+
+	if(custom_threat == "No")
+		return
+
 	if(!check_allowed(user))
 		return
 
-	declaring_war = 1
-	var/war_declaration = "[user.real_name] has declared his intent to utterly destroy [station_name()] with a nuclear device, and dares the crew to try and stop them."
+	war_declaration = stripped_input(user, "Insert your custom declaration", "Declaration")
```

Comment:
You should probably add `max_length=MAX_BROADCAST_LEN` as an argument to stripped_input here, so the declaration is limited to 512 characters instead of 1024